src/rules/awaitPromiseRule.ts

```diff
@@ -34,41 +43,43 @@ export class Rule extends Lint.Rules.TypedRule {
 
     public static FAILURE_STRING = "'await' of non-Promise.";
 
-    public applyWithProgram(srcFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
-        return this.applyWithWalker(new Walker(srcFile, this.getOptions(), program));
+    public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
+        const promiseTypes = new Set(["Promise", ...this.ruleArguments.map(String)]);
```

Comment:
If the config contains non-string values, does it really matter to convert them to string at runtime? Without explicitly converting them, they'll simply be ignored, because they are wrong either way.